### ISSUE TITLE
Issues setting up with .graphql file

### ISSUE THREAD

**Issue by ivansard** [dotansimha/graphql-code-generator#3758]
**Describe the bug**
I'm having issues generating the appropriate type definitions with any type of file that is stated in the docs. The only way I am able to successfully generate type definitions is if the value of the schema property is a URL to my running GraphQL server. I don't like this solution and would want to achieve this with any of the other 3 options (.graphl, .ts and .json). For this case, let's try with the .graphql file type

1. My GraphQL schema (schema.graphql):

```graphql
directive @auth on FIELD_DEFINITION
directive @admin on FIELD_DEFINITION

# QUERIES
type Query {
    getHomeFolders(data: GetHomeFoldersInput): FoldersPaginationOutput @auth

    getFolderContent(data: GetFolderContentInput): FolderContent

    findFolders(data: FindFoldersInput!): FoldersPaginationOutput!

    # Message

    getMessage(data: GetMessageInput): Message

    getStarredMessagesGlobal: [FolderContentGlobal!]!

    getCheckboxMessagesGlobal: [FolderContentGlobal]!

    findMessages(data: FindMessagesInput!): MessagesPaginationOutput! @auth

    filterMessages(data: FilterMessagesInput!): MessagesPaginationOutput!

    # User

    getUserById(id: ID!): User!

    me: User!

    getAllUsers(data: GetAllUsersInput!): UserPaginationOutput! @admin

    findUsers(data: FindUsersInput!): UserPaginationOutput! @admin
}

# MUTATIONS
type Mutation {
    updateFolder(data: UpdateFolderInput!): Folder! @auth

    deleteFolders(data: DeleteFoldersInput!): Boolean! @auth

    createFolder(data: CreateFolderInput!): Folder! @auth

    reorderFolders(data: ReorderFoldersInput!): Boolean!

    moveFolder(data: MoveFolderInput!): Boolean!

    #  Message

    createMessage(data: CreateMessageInput!): Message! @auth

    updateMessage(data: UpdateMessageInput!): Message! @auth

    moveMessages(data: MoveMessagesInput!): Boolean! @auth

    deleteMessages(data: DeleteMessagesInput!): Boolean! @auth

    signS3(filename: String!, filetype: String!): S3Payload

    addCheckbox(data: AddCheckboxInput!): Message!

    removeCheckbox(data: RemoveCheckboxInput!): Message!

    checkMessage(data: CheckMessageInput!): Message!

    uncheckMessage(data: UncheckMessageInput!): Message!

    starMessage(data: StarMessageInput!): Message!

    unstarMessage(data: UnstarMessageInput!): Message!

    # User

    createAdmin(data: CreateAdminInput!): Boolean!

    register(data: RegisterInput!): AuthPayload!

    login(data: LoginInput!): AuthPayload!

    forgotPasswordMail(data: ForgotPasswordInput!): Boolean!

    resetPasswordMail(data: ResetPasswordMailInput!): Boolean!

    updateUserDetails(data: UpdateUserDetailsInput!): User! @auth

    lockAccountManual(data: LockAccountManualInput!): Boolean! @auth

    resetPasswordManual(data: ResetPasswordManualInput!): Boolean! @auth

    deleteAccount(data: DeleteAccountInput!): Boolean! @auth

    verifyAccount(data: VerifyAccountInput!): Boolean! @admin

    fixDb: Boolean!
}

# SUBSCRIPTIONS
extend type Subscription {
    newFolder: Folder

    # Message

    newMessage(folderId: String): Message
}

# TYPES
type Folder {
    id: ID!
    title: String
    owner: String
    before: ID
    messages: [Message!]
    parent: ID
}

type FolderContent {
    folders: [Folder!]!
    messages: MessagesPaginationOutput!
}

type FoldersPaginationOutput {
    folders: [Folder!]!
    cursor: String
}

type Message {
    id: ID!
    type: String
    body: String
    folder: Folder
    image: Media
    title: String
    owner: String
    file: Media
    createdAt: Date
    updatedAt: Date
    cursor: String
    isCheckbox: Boolean
    isChecked: Boolean
    isStarred: Boolean
    urlMeta: [UrlMeta!]!
}

type S3Payload {
    signedRequest: String!
    url: String!
}

type MessagesPaginationOutput {
    messages: [Message!]!
    cursor: String
}

type FolderContentGlobal {
    folder: Folder!
    messages: [Message!]!
}

type Media {
    url: String!
    name: String
    type: String!
    size: Int!
}

type UrlMeta {
    author: String
    date: String
    description: String
    image: String
    logo: String
    publisher: String
    title: String
    url: String
}

enum MessageType {
    STAR
    CHECKBOX
    IMAGE
    FILE
    TEXT
}

type User {
    id: ID!
    email: String
    fullName: String
    folders: [Folder!]
    role: UserRole
    createdAt: Date
    updatedAt: Date
    cursor: String
}

type UserPaginationOutput {
    users: [User!]!
    cursor: String
}

type AuthPayload {
    token: String!
    refreshToken: String!
}

enum UserRole {
    ADMIN
    GUEST
}

# INPUTS

enum SortFields {
    createdAt
    title
}

enum SortOrder {
    ASC
    DESC
}

input DeleteFoldersInput {
    folders: [ID!]!
}

input UpdateFolderInput {
    folderId: ID!
    title: String!
}

input CreateFolderInput {
    title: String!
    parent: ID
    before: ID
}

input GetFolderContentInput {
    folderId: ID!
    messageCursor: String
}

input ReorderFoldersObj {
    id: ID!
    before: ID
}

input ReorderFoldersInput {
    folderPrimary: ReorderFoldersObj!
    folderFollower: ReorderFoldersObj
    folderDestination: ReorderFoldersObj
}

input MoveFolderInput {
    folderToMove: ID!
    newParent: ID!
}

input FindFoldersInput {
    text: String!
}

input GetHomeFoldersInput {
    cursor: String
}

input MediaInput {
    url: String!
    name: String
    type: String!
    size: Int!
}

input UrlMetaInput {
    author: String
    date: String
    description: String
    image: String
    logo: String
    publisher: String
    title: String
    url: String
}

input GetMessageInput {
    id: ID!
}

input MessageContent {
    title: String
    body: String
    image: MediaInput
    file: MediaInput
    urlMeta: UrlMetaInput
}

input CreateMessageInput {
    folderId: ID!
    content: MessageContent
}

input MoveMessagesInput {
    toFolder: ID!
    messages: [ID!]!
}

input UpdateMessageInput {
    messageId: ID!
    content: MessageContent
}

input DeleteMessagesInput {
    messages: [ID!]!
}

input FindMessagesInput {
    text: String!
    folder: ID
    cursor: String
}

input AddCheckboxInput {
    messageId: ID!
}

input RemoveCheckboxInput {
    messageId: ID!
}

input CheckMessageInput {
    messageId: ID!
}

input UncheckMessageInput {
    messageId: ID!
}

input StarMessageInput {
    messageId: ID!
}

input UnstarMessageInput {
    messageId: ID!
}

input FilterMessagesInput {
    type: MessageType!
    folder: ID!
    cursor: String
}

input RegisterInput {
    email: String!
    password: String!
    fullName: String!
}

input LoginInput {
    email: String!
    password: String!
}

input ForgotPasswordInput {
    email: String!
}

input ResetPasswordMailInput {
    token: String!
    newPassword: String!
}

input UpdateUserDetailsInput {
    fullName: String!
}

input CreateAdminInput {
    email: String!
    password: String!
    fullName: String!
}

input FindUsersInput {
    text: String!
    cursor: String
}

input GetAllUsersInput {
    cursor: String
}

input LockAccountManualInput {
    email: String!
}

input ResetPasswordManualInput {
    newPassword: String!
}

input DeleteAccountInput {
    email: String!
}

input VerifyAccountInput {
    email: String!
}


```

2. My `codegen.yml` config file:

```yml
overwrite: true
schema: "schema.graphql"
documents: null
generates:
  src/generated/graphql.ts:
    plugins:
      - "typescript"
      - "typescript-resolvers"
  ./graphql.schema.json:
    plugins:
      - "introspection"

```
**Expected behavior**

When I run `npm run generate` which runs `graphql-codegen --config codegen.yml`, the script returns an error, with the message 'Something went wrong'. The same schema when fetched from a remote graphql server successfully generates the type definitions

**Environment:**

- OS: Ubuntu 18.04
- `@graphql-codegen/...`:   1.13.1
- NodeJS: 13.10.1

**Comment by urbgimtam** [comment]
Have you tried downgrading to version 1.13.0?
I've started to get a similar error when upgraded to version 1.13.1, but 1.13.0 works perfectly.

**Comment by ivansard** [comment]
Hey @urbgimtam, thank you very much for the reply. 

I have downgraded now to version 1.13.0, and I am still getting the same error. I have also tried with a schema.json file, which contains the introspection JSON from my schema, which I queried on my remote server and copied into the file. The query fired in the playground is below:

```
query {
  __schema {
    types {
      name
      kind
      description
      fields {
        name
      }
    }
  }
}
```

And I have adjusted my codegen.yml as so:

```
overwrite: true
schema: "./schema.json"
documents: null
generates:
  src/generated/graphql.ts:
    plugins:
      - "typescript"
      - "typescript-resolvers"
  ./graphql.schema.json:
    plugins:
      - "introspection"

```

This is also not working.

Could you please send me your configuration, so that I can maybe see something that I am missing?

Thanks once again for the reply and your time

**Comment by dotansimha** [comment]
The actual issue here is related to the way we display the error messages, for unknown reason it doesn't show the error correctly. I will investigate it. 

In your schema, you are using a type called `Date` without having it defined. It's missing `scalar Date`, I guess you are using an external library for that? if so, note that you need to tell codegen it exists as well.

**Comment by dotansimha** [comment]
Fixed by @kamilkisiela in: https://github.com/dotansimha/graphql-code-generator/pull/3816

**Comment by ivansard** [comment]
@dotansimha thanks for the reply. 

It was down to the Date scalar missing from the .graphql file. I've successfully generated type with both the .graphql file as well as with the .json file.

Thank you very much for your help, looking forward to further releases of the library, you guys are doing a fantastic job!

Cheers

**Comment by dotansimha** [comment]
Fixed in v1.13.2
